services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/testdata/TestDataServiceITest.kt

```diff
@@ -0,0 +1,77 @@
+package fi.hel.haitaton.hanke.testdata
+
+import assertk.assertThat
+import assertk.assertions.each
+import assertk.assertions.hasSize
+import assertk.assertions.isNull
+import assertk.assertions.isTrue
+import fi.hel.haitaton.hanke.DatabaseTest
+import fi.hel.haitaton.hanke.allu.ApplicationStatus
+import fi.hel.haitaton.hanke.application.ApplicationRepository
+import fi.hel.haitaton.hanke.factory.AlluDataFactory
+import fi.hel.haitaton.hanke.factory.HankeFactory
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.context.ActiveProfiles
+import org.testcontainers.junit.jupiter.Testcontainers
+
+private const val USERNAME = "testUser"
+
+@Testcontainers
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = ["haitaton.testdata.enabled=true"],
+)
+@ActiveProfiles("default")
+@WithMockUser(USERNAME)
+class TestDataServiceITest : DatabaseTest() {
+
+    @Autowired private lateinit var testDataService: TestDataService
+    @Autowired private lateinit var applicationRepository: ApplicationRepository
+    @Autowired private lateinit var alluDataFactory: AlluDataFactory
+    @Autowired private lateinit var hankeFactory: HankeFactory
+
+    @Nested
+    inner class UnlinkApplicationsFromAllu {
+        @Test
+        fun `Without applications does nothing`() {
```

Comment:
Does this actually verify that it does not do anything?

Reply:
The important thing is that it doesn't throw an exception with an empty list. I renamed the test.